docs/source/DeveloperGuide/OAuth2Authorization/index.rst

```diff
@@ -0,0 +1,281 @@
+
+.. note:: to use the received access token to access DIRAC services, you need to add ``/DIRAC/Security/UseTokens=true`` or ``export DIRAC_USE_ACCESS_TOKEN=true``.
+
+
+**********
+Logging in
+**********
+
+Consider process by which an user gains access to a DIRAC resources by identifying and authenticating themselves.
+
+DIRAC CLI
+=========
+
+The ``dirac-login`` command will help us with this. There are three main ways to authorize:
+
+- using a local user certificate to obtain a proxy certificate
+- logging in with DIRAC AS to obtain a proxy certificate
+- logging in with DIRAC AS to obtain an access token
+
+
+Using ``dirac-login my_group --use-certificate``:
+
+.. image:: /_static/Systems/FS/OAuth2/certificateFlow.png
+   :alt: DIRAC CLI login with certificate flow.
+
+Using the local certificate ``dirac-login`` makes a similar algorithm as :ref:`dirac-proxy-init`:
+  1) Generate a proxy certificate locally on the user's machine from a locally installed user certificate.
+  #) Try to connect to the DIRAC Configuration Server (CS) with this proxy certificate.
+  #) If the connection was successful, a command generate a proxy certificate with the required extensions.
+  #) A proxy certificate without extensions upload to :py:class:`~DIRAC.FrameworkSystem.DB.ProxyDB.ProxyDB` using :py:class:`~DIRAC.FrameworkSystem.Service.ProxyManagerHandler.ProxyManagerHandler`.
+
+Using ``dirac-login my_group --use-diracas --token``:
+
+.. image:: /_static/Systems/FS/OAuth2/diracasTokenFlow.png
+   :alt: DIRAC CLI login DIRAC AS flow and obtaining an access token.
+
+User do not need to have a locally installed certificate if logging in through DIRAC AS.
+
+OAuth 2.0 Device flow:
+  1) ``dirac-login`` initializes `OAuth 2.0 Device flow` by passing DIRAC client ID to DIRAC AS.
+  #) DIRAC AS responds with a ``device_code``, ``user_code``, ``verification_uri``, ``verification_uri_complete``, ``expires_in`` (lifetime in seconds for device_code and user_code), and polling ``interval``.
+  #) The command asks the user to log in using a device that has a browser(e.g.: their computer, smartphone) or if the device running ``dirac-login`` has a browser installed, a new tab with the received URL will open automatically.
+
+    a) The command begins polling DIRAC AS for an access token sending requests to token endpoint until either the user completes the browser flow path or the user code expires.
+
+OAuth 2.0 Authorization Code flow:
+  4) After receiving this request from the browser, DIRAC AS will initialize ``OAuth 2.0 Authorization Code`` flow with choosed IdP. If several IdPs are registered in DIRAC and it is not clear from the requested group which one to choose, DIRAC AS will ask the user to choose one.
```

Comment:
shouldn't this start with 1)?

Reply:
I changed the structure a bit to look holistic in one context